osquery/tables/system/windows/userassist.cpp

```diff
@@ -0,0 +1,143 @@
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+#include <osquery/tables/system/windows/registry.h>
+#include <osquery/utils/conversions/split.h>
+#include <osquery/utils/system/time.h>
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+// Decode ROT13 sub key value
+std::string rot_decode(std::string value_key_reg) {
+  std::string decoded_value_key;
+
+  for (size_t i = 0; i < value_key_reg.size(); i++) {
+    if (isalpha(value_key_reg[i])) {
+      if (value_key_reg[i] >= 'a' && value_key_reg[i] <= 'm') {
+        decoded_value_key.append(1, value_key_reg[i] + 13);
+      } else if (value_key_reg[i] >= 'm' && value_key_reg[i] <= 'z') {
+        decoded_value_key.append(1, value_key_reg[i] - 13);
+      } else if (value_key_reg[i] >= 'A' && value_key_reg[i] <= 'M') {
+        decoded_value_key.append(1, value_key_reg[i] + 13);
+      } else if (value_key_reg[i] >= 'M' && value_key_reg[i] <= 'Z') {
+        decoded_value_key.append(1, value_key_reg[i] - 13);
+      }
+    } else {
+      decoded_value_key.append(1, value_key_reg[i]);
+    }
+  }
+  return decoded_value_key;
+}
+
+QueryData genUserAssist(QueryContext& context) {
+  QueryData results;
+  QueryData users;
+
+  queryKey("HKEY_USERS", users);
+
+  for (const auto& uKey : users) {
+    if (uKey.count("type") == 0 || uKey.count("path") == 0) {
+      continue;
+    }
+
+    std::string path = uKey.at("path");
+    std::string fullPath = path.append(
+        "\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\UserAssist");
+
+    QueryData user_assist_results;
+    queryKey(fullPath, user_assist_results);
+
+    for (const auto& rKey : user_assist_results) {
+      if (rKey.count("type") == 0 || rKey.count("path") == 0) {
+        continue;
+      }
+      std::string path_key = rKey.at("path");
+      std::string full_path_key = path_key.append("\\Count");
+
+      QueryData assist_results;
+      queryKey(full_path_key, assist_results);
+
+      for (const auto& aKey : assist_results) {
+        std::string subkey = aKey.at("path");
+
+        // split reg path by \Count\ to get Key values
+        std::size_t count_key = subkey.find("Count\\");
+        std::string value_key = subkey.substr(count_key);
+        std::string value_key_reg = value_key.substr(6, std::string::npos);
+
+        std::string decoded_value_key = rot_decode(value_key_reg);
+
+        Row r;
+
+        // set UEME_CTLCUACount:ctor and UEME_CTLSESSION values to blank they
+        // are not executables
+        if (decoded_value_key == "UEME_CTLCUACount:ctor" ||
+            decoded_value_key == "UEME_CTLSESSION") {
+          r["path"] = decoded_value_key;
+          r["last_execution_time"] = "";
+          r["count"] = INTEGER("");
+          r["sid"] = uKey.at("name");
+          results.push_back(r);
+        } else {
+          std::string assist_data = aKey.at("data");
```

Comment:
This is some complex logic (in the `else` statement), I recommend breaking it out into a separate method and adding a test for that.